go/integration_test/processor/processor_gas_test.go

```diff
@@ -145,132 +143,225 @@ func getGasTestScenarios() map[string]Scenario {
+		"SimpleCodeInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
 				},
 			},
-		},
-		Transaction: tosca.Transaction{
-			Sender:    tosca.Address{1},
-			Recipient: &tosca.Address{2},
-			GasLimit:  floria.TxGas + executionGasCost - 1,
-			Nonce:     4,
-		},
-		After: WorldState{
-			{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
-			{2}: Account{Balance: tosca.NewValue(0),
-				Code: tosca.Code{
-					byte(op.PUSH1), byte(0), // < PUSH 0
-					byte(op.PUSH1), byte(0), // < PUSH 0
-					byte(op.RETURN),
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + executionGasCost - 1,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
 				},
 			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: floria.TxGas + executionGasCost - 1,
+			},
+			OperaError: fmt.Errorf("gas too low"),
 		},
-		Receipt: tosca.Receipt{
-			Success: false,
-			GasUsed: floria.TxGas + executionGasCost - 1,
+		"InternalCallDoesNotConsume10PercentOfRemainingGas": {
+			Before: WorldState{
+				{}:  Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + 100,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{}:  Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas,
+			},
 		},
-		OperaError: fmt.Errorf("gas too low"),
 	}
 
-	allTestCases["InternalCallDoesNotConsume10RemainingPercentGas"] = Scenario{
-		Before: WorldState{
-			{}:  Account{Balance: tosca.NewValue(100), Nonce: 4},
-			{2}: Account{Balance: tosca.NewValue(0)},
-		},
-		Transaction: tosca.Transaction{
-			Sender:    tosca.Address{},
-			Recipient: &tosca.Address{2},
-			GasLimit:  floria.TxGas + 100,
-			Nonce:     4,
+	return testCases
+}
+
+func gasPricingTestCases() map[string]Scenario {
+	gasPrice := uint64(10)
+	testCases := map[string]Scenario{
+		"GasPriceCalculation": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(floria.TxGas * gasPrice), Nonce: 4},
+				{2}: Account{},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas,
+				GasPrice:  tosca.ValueFromUint256(uint256.NewInt(gasPrice)),
```

Comment:
Here and at other places:
```suggestion
				GasPrice:  tosca.NewValue(gasPrice),
```